diga-epa-client/src/main/java/com/oviva/epa/client/internal/KonnektorServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+import com.oviva.epa.client.internal.svc.phr.model.RecordIdentifier;
+import com.oviva.epa.client.internal.svc.phr.model.RecordIdentifierAdapter;
+import com.oviva.epa.client.konn.KonnektorConnection;
+import com.oviva.epa.client.model.WriteDocumentException;
+import de.gematik.epa.LibIheXdsMain;
+import de.gematik.epa.ihe.model.document.Document;
+import de.gematik.epa.ihe.model.request.DocumentSubmissionRequest;
+import de.gematik.epa.ihe.model.simple.AuthorInstitution;
+import de.gematik.epa.ihe.model.simple.SubmissionSetMetadata;
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import oasis.names.tc.ebxml_regrep.xsd.rs._3.RegistryErrorList;
+import oasis.names.tc.ebxml_regrep.xsd.rs._3.RegistryResponseType;
+import telematik.ws.conn.cardservicecommon.xsd.v2_0.CardTypeType;
+
+public class KonnektorServiceImpl implements KonnektorService {
+
+  private static final String STATUS_SUCCESS =
+      "urn:oasis:names:tc:ebxml-regrep:ResponseStatusType:Success";
+  private final EventServiceClient eventServiceClient;
+  private final CardServiceClient cardServiceClient;
+  private final PhrManagementServiceClient phrManagementServiceClient;
+  private final PhrServiceClient phrServiceClient;
+
+  private final SmbInformationServiceClient smbInformationServiceClient;
+
+  public KonnektorServiceImpl(KonnektorConnection connection, KonnektorContext konnektorContext) {
+
+    eventServiceClient = new EventServiceClient(connection.eventService(), konnektorContext);
+
+    cardServiceClient =
+        new CardServiceClient(connection.cardService(), konnektorContext, eventServiceClient);
+
+    phrServiceClient = new PhrServiceClient(connection.phrService(), konnektorContext);
+
+    phrManagementServiceClient =
+        new PhrManagementServiceClient(connection.phrManagementService(), konnektorContext);
+
+    var certificateServiceClient =
+        new CertificateServiceClient(connection.certificateService(), konnektorContext);
+
+    smbInformationServiceClient =
+        new SmbInformationServiceClient(eventServiceClient, certificateServiceClient);
+  }
+
+  @Override
+  public List<AuthorInstitution> getAuthorInstitutions() {
+    return smbInformationServiceClient.getAuthorInstitutions();
+  }
+
+  @Override
+  public List<Card> getCardsInfo() {
+    // don't be fooled by the name, this returns a list of cards...
+    return eventServiceClient.getSmbInfo().getCards().getCard().stream()
+        .map(c -> new Card(c.getCardHandle(), c.getCardHolderName(), mapCardType(c.getCardType())))
+        .toList();
+  }
+
+  private CardType mapCardType(CardTypeType t) {
+    return switch (t) {
+      case SMC_B -> CardType.SMC_B;
+      case SMC_KT -> CardType.SMC_KT;
+      default -> CardType.UNKNOWN;
+    };
+  }
+
+  @Override
+  public PinStatus verifyPin(String cardHandle) {
+    var response = cardServiceClient.getPinStatusResponse(cardHandle, "PIN.SMC");
```

Comment:
you may want to differentiate here as well, I don't think we will ever be interested in other PINs but who knows 😄

Reply:
Done, renamed